docs site documentation on installing airgapped Kommander (#3221)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.1/install-airgapped/index.md

---
layout: layout.pug
navigationTitle: Install air-gapped
title: Install Kommander air-gapped
menuWeight: 35
excerpt: Install Kommander in an air-gapped environment
enterprise: false
---

This document is meant to outline steps required to install kommander in an airgapped environment. It uses documentation of airgapped Konvoy installation as basis and outlines only the steps required to get Kommander running on top of an airgapped Konvoy cluster.

# Naming

Across this document, some terms are used which are defined here:
* management cluster - konvoy cluster that is going to be running kommander
* attached cluster - cluster, be it konvoy or non-konvoy one, which are going to be attached to the management cluster
* docker registry - a registry from which all clusters download docker images used by installation

# Before you begin

Before installing, verify that your environment meets the following basic requirements:

* a docker registry should already contain all the docker images required for installation and this includes also kommander images. The `konvoy_air_gapped.tar.bz2` tarball will contain the required artifacts.

> * connectivity with clusters that are going to be attached to the management

Reviewer: yes, the internal LBs created in the attached clusters, especially those of the federated resources, might need to be accessible from the management cluster.

  cluster:
  * both management and attached clusters should be able to connect to the docker registry

>   * management cluster should be able to connect to attached cluster's API server

Reviewer: as mentioned before, e.g. thanos needs to talk to the internal LBs exposed as internal LBs in the attached clusters.

Author: Added new bulletpoint:
```
* management cluster must connect to load balancers created by some of the addons. An example of such set up is Thanos which is part of prometheus addon
```

* all the prerequisites outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#before-you-begin] in case of konvoy clusters. For attached clusters that are 3rd party clusters, please refer to the vendor's documentation.

## Control plane nodes

> Control plane nodes of Konvoy clusters should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters control plance requirements refer to the vendor specification.

Reviewer 2: ```suggestion
Konvoy cluster control plane nodes must meet the minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters and their control plane requirements, refer to the vendor specification.
```

> Control plane nodes of Konvoy clusters should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#control-plane-nodes]. For 3rd-party clusters control plance requirements refer to the vendor specification.

Reviewer 3: typo:  "plance" --> "plane"

## Worker nodes

> Worker nodes should meet the minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#worker-nodes].

Reviewer 2: ```suggestion
Worker nodes must meet the minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#worker-nodes].
```

## Operating system and services for all nodes

> All nodes should meet the same minimal requirements outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes].

Reviewer 2: ```suggestion
All nodes must meet the same minimal requirements outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#operating-system-and-services-for-all-nodes].
```

> ## Defininig inventory file

Reviewer: typo Defininig

> ## Defininig inventory file

Reviewer 2: ```suggestion
## Definine the inventory file
```

> Installing airgapped Kommander does not require any additional changes in the `inventory.yaml` file.

Reviewer 2: ```suggestion
Installing air-gapped Kommander does not require any changes in the `inventory.yaml` file.
```

# Configure the Kubernetes cluster

> The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an airgapped environment requires some additional configuration, which is described below. It is assumed that all the `cluster.yaml` changes outlined in [airgapped konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation were also applied.

Reviewer 2: ```suggestion
The `cluster.yaml` file provides the configuration details for creating your Konvoy cluster. Installing Kommander in an air-gapped environment requires additional configuration. The following steps describe these changes. Ensure all the `cluster.yaml` changes outlined in [air-gapped Konvoy installation][https://docs.d2iq.com/ksphere/konvoy/1.6/install/install-airgapped/#configure-the-image-registry] documentation are applied.
```

> First of all, Kommander needs to use self-hosted charts repo that will be running on top of Konvoy cluster, as it will not be able to connect to the default one through the public Internet.

Reviewer: nit: `repository`

> First of all, Kommander needs to use self-hosted charts repo that will be running on top of Konvoy cluster, as it will not be able to connect to the default one through the public Internet.

Reviewer 2: ```suggestion
1.  Ensure Kommander can use the self-hosted charts repo running on top of the Konvoy cluster. It can not connect to the default one through the public Internet.
```

> First of all, Kommander needs to use self-hosted charts repo that will be running on top of Konvoy cluster, as it will not be able to connect to the default one through the public Internet.

Reviewer 3: nit: "on top of Konvoy cluster" -> "on top of the Konvoy cluster"

```yaml
- name: kommander
  values: |2
    global:
      federate:
        airgapped:
          enabled: true
          chartRepo: http://konvoy-addons-chart-repo.kubeaddons.svc:8879
    kommander-federation:
      clusterAutoscaler:
        chartRepo: http://konvoy-addons-chart-repo.kubeaddons.svc:8879
```

> Secondly, Kommander needs to know where the private docker registry is and how to access it. The `registry_ip` variable is the ip address on which private docker registry is available, ${user} and ${password} are username and password if credentials are required.

Reviewer 2: ```suggestion
1.  Ensure  Kommander can find and access the private Docker registry. The `registry_ip` variable is the ip address of the available private Docker registry. ${user} and ${password} are the username and password if access requires credentials.
```

```yaml
- name: kommander
  values: |2
    kommander-federation:
      konvoy:
        imageRepository: "${registry_ip}:5000/mesosphere/konvoy"
      utilityApiserver:
        extraArgs:
          docker-registry-url: "https://${registry_ip}:5000"
          docker-registry-insecure-skip-tls-verify: true
          docker-registry-username: 'admin'
          docker-registry-password: 'password'
```

> And finally, we kommander controller needs to be reconfigured to work in an airgapped environment:

Reviewer 2: ```suggestion
1.  Reconfigure the Kommander controller to work in an air-gapped environment:
```

```yaml

> - name: kommander

Reviewer: I personally don't like this fragmented explanation. I prefer to explain it all together.

Author: I am fine both ways. @russurquhart1 WDYT?

Reviewer 4: I agree with Hector.

Author: Done in commit `Remove explaining kommander configuration via snippets`

  values: |2

>     kommander-federation:

Reviewer: you also need to link this change too https://github.com/mesosphere/dcos-docs-site/pull/3221/files#diff-ed1b6690070eb1879058e863d81372abR103 as part of enabling airgapped for kommander

Author: Done here: https://github.com/mesosphere/dcos-docs-site/pull/3221/files#diff-ed1b6690070eb1879058e863d81372abR54

      controller:
        containers:
          manager:
            extraArgs:
              feature-gates: "Airgapped=true"
```

> Putting it together, the Kommander addon configuration should look like this:

Reviewer 2: ```suggestion
    The Kommander addon configuration should look similar to the following:
```
